### PR TITLE
feat(coverage): add summary reporter

### PR DESCRIPTION
**Pull request by kt3k**
This PR adds 'summary' reporter to `deno coverage` command. The reporter shows the coverage report in the terminal like the below:

Screenshot 2023-12-11 at 21 55 28

This change also makes this output format default of `deno coverage`. The existing default output (PrettyCoverageReporter) is now available via `--pretty` flag. (@bartlomieju and I discussed about this default behavior change offline).

part of #21325

### CODE REVIEW

File: cli/tests/integration/coverage_tests.rs [hunk 1]
```diff
@@ -575,3 +590,42 @@ fn test_html_reporter() {
...
+    .new_command()
+    .args_vec(vec!["coverage".to_string(), format!("{}/", tempdir)])
+    .run();
+
+  output.assert_exit_code(0);
+  output.assert_matches_text(
+    "----------------------------------
+File         | Branch % | Line % |
+----------------------------------
+ All files   |    40.00 |  60.98 |
```
Comment (bartlomieju) [inline cli/tests/integration/coverage_tests.rs:623]: Does this output match one from any of the existing tools?

I'm thinking that maybe we could do something like this:
```
----------------------------------
File         | Branch % | Line % |
----------------------------------
 bar.ts      |     0.00 |  57.14 |
 baz/quux.ts |     0.00 |  28.57 |
 baz/qux.ts  |   100.00 | 100.00 |
 foo.ts      |    50.00 |  76.92 |
----------------------------------
 All files   |    40.00 |  60.98 |
----------------------------------
```

Comment (kt3k) [reply cli/tests/integration/coverage_tests.rs]: Updated. Looks better to me

Screenshot 2023-12-12 at 0 15 45

Comment (ry) [reply cli/tests/integration/coverage_tests.rs]: Probably one decimal place of precision would be enough.

Comment (kt3k) [reply cli/tests/integration/coverage_tests.rs]: Updated

Screenshot 2023-12-12 at 1 05 34

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**